docs/getting-started/part-3/incremental-mode.md

```diff
@@ -0,0 +1,132 @@
+```
+Use now this slightly modified command to run the `deduplicate-departures` feed. Nothing should have changed so far, since we only read and write an empty state. 
+You can can check that by opening the file `getting-started.<runId>.<attemptId>.json` and having a look at the field `dataObjectsState`. The stored state is currently empty. 
+In the next section we will assign a value to `nextState`, such that the is `dataObjectsState` is getting written. 
+The two variables `<runId>` and `<attemptId>` describe smart data like intrinsics. 
+For each execution the `<runId>` is incremented by one. The `<attemptId>` is normally 1 and only increased by one if smart data lake builder had to recover a failed execution. This recovery mechanism will be part of a future tutorial.
+
+## Define a Query Logic
+What we would like to achieve is the following query logic:
+
+The starting point is the query parameters provided in the configuration file and no previous state. 
+During the first execution, we query the departures for the two airports in the given time window. 
+Afterwards, we store for each airport the `begin`-parameter for the next query. This equals the `end`-parameter of the current query. 
+Now the true incremental phase starts as we now have stored the state for the next starting point. We query the API from the `begin` stored in the previous state until now. 
+For this to work, we need to make two changes. First add the variable
+```
+private val now = Instant.now.getEpochSecond
+
+``` 
+just below the `nextState` variable. Then modify the `currentQueryParameters` variable according to
+```scala
+// if we have query parameters in the state we will use them from now on
+val currentQueryParameters = if (previousState.isEmpty) queryParameters.get else previousState.map{
+  x => DepartureQueryParameters(x.airport, x.nextBegin, now)
+}
+```
+and move it below the comment `// place the new implementation of currentQueryParameters below this line`, which can be found in the `getDataFrame` method. The implemented logic 
+```scala
+if(previousState.isEmpty){
+  nextState = currentQueryParameters.map(params => State(params.airport, params.end))
+} else {
+  nextState = previousState.map(params => State(params.airport, now))
+}
+```
+for the next state can be placed below the comment `// put simple nextState logic below`. Now you should again build the docker image and run it multiple times. The scenario will be that the first run fetches the data defined in the configuration file, then the proceeding run retrieves the data from the endpoint of the last run until now. And finally the third execution will most probably fail, as only little seconds have been passed and most likely no data is available in such a short time window. Sadly, the api response in such cases with a **404** error code instead with a **200** and an empty response.
```

Comment:
```suggestion
for the next state can be placed below the comment `// put simple nextState logic below`. Now you should again build the docker image and run it multiple times. The scenario will be that the first run fetches the data defined in the configuration file, then the proceeding run retrieves the data from the endpoint of the last run until now. And finally the third execution will most probably fail, as only little seconds have been passed and most likely no data is available in such a short time window. Unfortunately, the webservice on opensky-network.org responds with a **404** error code when no data is available, rather than a **200** and an empty response. Therefore, SDL gets a 404 and will fail the execution.
```